Test/Sampler/test_sampler.py

```diff
@@ -0,0 +1,154 @@
+import netket as nk
+import networkx as nx
+import numpy as np
+import pytest
+from mpi4py import MPI
+from pytest import approx
+
+samplers = {}
+
+
+# TESTS FOR SPIN HILBERT
```

Comment:
@twesterhout these tests apparently give a seg fault on linux machines ONLY. Is this trivial as well?!? ! Anyway, these memory management issues are a mess.

Reply:
Okay, this one is slightly more complicated, because I had to look through a lot of code, but _yes_, that's again the exact same thing! :) See the other comment for the explanation